docker-app/qfieldcloud/core/rest_utils.py

```diff
@@ -24,19 +23,17 @@ def exception_handler(exc, context):
     elif isinstance(exc, exceptions.ValidationError):
         qfc_exc = qfieldcloud_exceptions.ValidationError(detail=str(exc))
     elif isinstance(exc, qfieldcloud_exceptions.QFieldCloudException):
-        is_error = True
         qfc_exc = exc
     elif isinstance(exc, rest_exceptions.APIException):
-        is_error = True
         qfc_exc = qfieldcloud_exceptions.APIError(exc.detail, exc.status_code)
     else:
         # Unexpected ! We rethrow original exception to make debugging tests easier
         if settings.IN_TEST_SUITE:
             raise exc
-        is_error = True
         qfc_exc = qfieldcloud_exceptions.QFieldCloudException(detail=str(exc))
 
-    if is_error:
+    # Log level is defined by the exception
+    if qfc_exc.log_as_error:
```

Comment:
From what I see, it seems only errors are logged to sentry. Would it make sense to also log info (which maybe we could promote to warning) for user caused issues ? I guess they could sometime indicate serious issues still (e.g. broken links, bugs in QField, etc).

Not sure how this would be done, found this: https://docs.sentry.io/platforms/python/guides/django/usage/set-level/

Also not sure where and how we use logging, just so that we don't fill sentry's servers with junk...

We also have settings.LOGLEVEL in our settings.py, but I think that is unused code ?

Reply:
Of course wasn't talking about logging debug messages to sentry... Did you check how many info messages there is ? At least just warnings would already be good, as it would include stuff like deprecation warning or unclosed resources, which are definitely worth knowing of. There definitely should not be too many of them (or if there are, we better know about them).

Or are we somehow checking the logs in a different way to find out about such issues ?